iot_verifier/src/gateway_updater.rs

```diff
@@ -0,0 +1,95 @@
+use crate::Settings;
+use chrono::{Duration, Utc};
+use futures::stream::StreamExt;
+use helium_crypto::PublicKeyBinary;
+use iot_config::{
+    client::{Client as IotConfigClient, ClientError as IotConfigClientError},
+    gateway_info::{GatewayInfo, GatewayInfoResolver},
+};
+use std::collections::HashMap;
+use tokio::sync::watch;
+use tokio::time;
+
+pub type MessageSender = watch::Sender<GatewayMap>;
+pub type MessageReceiver = watch::Receiver<GatewayMap>;
+
+pub struct GatewayStore {
+    iot_config_client: IotConfigClient,
+    refresh_interval: Duration,
+    pub receiver: MessageReceiver,
+    sender: MessageSender,
+}
+
+pub struct GatewayMap {
+    pub gateways: HashMap<PublicKeyBinary, GatewayInfo>,
+    pub timestamp: i64,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum GatewayStoreError {
+    #[error("gateway not found: {0}")]
+    GatewayNotFound(PublicKeyBinary),
+    #[error("error querying iot config service")]
+    IotConfigClient(#[from] IotConfigClientError),
+}
+
+impl GatewayStore {
+    pub async fn from_settings(
+        settings: &Settings,
+        iot_config_client: IotConfigClient,
+    ) -> Result<Self, GatewayStoreError> {
+        let gateway_map = refresh_gateways(iot_config_client.clone()).await?;
+        let (sender, receiver) = watch::channel(gateway_map);
+        Ok(Self {
+            iot_config_client,
+            refresh_interval: settings.gateway_refresh_interval(),
+            receiver,
+            sender,
+        })
+    }
+
+    pub async fn run(&mut self, shutdown: &triggered::Listener) -> Result<(), GatewayStoreError> {
+        tracing::info!("starting gateway_store");
+
+        let mut trigger_timer = time::interval(
+            self.refresh_interval
+                .to_std()
+                .expect("valid interval in seconds"),
+        );
+
+        loop {
+            if shutdown.is_triggered() {
+                tracing::info!("stopping gateway_store");
+                return Ok(());
+            }
+
+            tokio::select! {
+                _ = trigger_timer.tick() => self.handle_refresh_tick().await?,
+                _ = shutdown.clone() => return Ok(()),
+            }
+        }
+    }
+
+    async fn handle_refresh_tick(&mut self) -> Result<(), GatewayStoreError> {
+        let updated_gateway_map = refresh_gateways(self.iot_config_client.clone()).await?;
+        _ = self.sender.send(updated_gateway_map);
+        Ok(())
+    }
+}
+
+pub async fn refresh_gateways(
+    iot_config_client: IotConfigClient,
```

Comment:
Change this to and you should be able to remove all the cloning of iot_config_client
```rust
iot_config_client: &mut IotConfigClient,
```

Reply:
addressed here: b8849cf44e3212e8159fd09c1135dc690073e002